dbms/src/Debug/MockKVStore/MockRaftStoreProxy.cpp

```diff
@@ -621,20 +623,27 @@ std::tuple<RegionPtr, PrehandleResult> MockRaftStoreProxy::snapshot(
         }
     }
     SSTViewVec snaps{ssts.data(), ssts.size()};
-    auto prehandle_result = kvs.preHandleSnapshotToFiles(new_kv_region, snaps, index, term, deadline_index, tmt);
-
-    auto rg = RegionPtrWithSnapshotFiles{new_kv_region, std::vector(prehandle_result.ingest_ids)};
-    if (cancel_after_prehandle)
+    try
     {
-        kvs.releasePreHandledSnapshot(rg, tmt);
+        auto prehandle_result = kvs.preHandleSnapshotToFiles(new_kv_region, snaps, index, term, deadline_index, tmt);
+        auto rg = RegionPtrWithSnapshotFiles{new_kv_region, std::vector(prehandle_result.ingest_ids)};
+        if (cancel_after_prehandle)
+        {
+            kvs.releasePreHandledSnapshot(rg, tmt);
+            return std::make_tuple(kvs.getRegion(region_id), prehandle_result);
+        }
+        kvs.checkAndApplyPreHandledSnapshot<RegionPtrWithSnapshotFiles>(rg, tmt);
+        // Though it is persisted earlier in real proxy, but the state is changed to Normal here.
+        region->updateAppliedIndex(index, true);
+        // Region changes during applying snapshot, must re-get.
         return std::make_tuple(kvs.getRegion(region_id), prehandle_result);
     }
-    kvs.checkAndApplyPreHandledSnapshot<RegionPtrWithSnapshotFiles>(rg, tmt);
-    // Though it is persisted earlier in real proxy, but the state is changed to Normal here.
-    region->updateAppliedIndex(index, true);
-
-    // Region changes during applying snapshot, must re-get.
-    return std::make_tuple(kvs.getRegion(region_id), prehandle_result);
+    catch (const Exception & e)
+    {
+        LOG_ERROR(log, e.message());
+        e.rethrow();
+    }
+    exit(0);
```

Comment:
should use `exit(1)` for abnormal exit? Or this is just unreachable?

Reply:
It's unreachable, but not worth a throw since it is only used in test codes.

Reply:
I think the `exit` line can be removed. If there is no exception, it directly return. If there is exception, then it will rethrow the exception.

Reply:
Then what do we return here? Since the compiler forces us to return something here

Reply:
got it, it should be replaced by
```suggestion
    catch (...)
    {
        tryLogCurrentFatalException(...);
        exit(1);
    }
```